app/src/main/java/agency/five/codebase/android/movieapp/ui/favorites/FavoritesScreen.kt

```diff
@@ -17,35 +14,30 @@ import androidx.compose.foundation.lazy.grid.items
 import androidx.compose.material.MaterialTheme
 import androidx.compose.material.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
 import androidx.compose.runtime.getValue
-import androidx.compose.runtime.mutableStateOf
-import androidx.compose.runtime.remember
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.res.dimensionResource
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.style.TextOverflow
 import androidx.compose.ui.tooling.preview.Preview
+import org.koin.androidx.compose.getViewModel
 
 private const val FAVORITES_GRID_COLUMNS_COUNT = 3
 
-private val favoritesMapper: FavoritesMapper = FavoritesMapperImpl()
-
-val movies = MoviesMock.getMoviesList()
-val favoriteMovies = movies.filter { it.isFavorite }
-
-val favoritesViewState = favoritesMapper.toFavoritesViewState(favoriteMovies)
-
 @Composable
-fun FavoritesRoute(onNavigateToMovieDetails: (Int) -> Unit) {
-    val favorites by remember {
-        mutableStateOf(favoritesViewState)
-    }
-    FavoritesScreen(favorites, onNavigateToMovieDetails)
+fun FavoritesRoute(
+    viewModel: FavoritesViewModel,
+    onNavigateToMovieDetails: (Int) -> Unit
+) {
+    val favoritesViewState: FavoritesViewState by viewModel.favoritesViewState.collectAsState()
+    FavoritesScreen(favoritesViewState, { viewModel.toggleFavorite(it) }, onNavigateToMovieDetails)
```

Comment:
you can pass a reference instead of creating lambda and invoking it, looks a bit nicer  -> `viewModel::toggleFavorite`